plugins/microservices/msi_get_open_data_obj_path/libmsi_get_open_data_obj_path.cpp

```diff
@@ -0,0 +1,83 @@
+/// \file
+
+#include "irods_ms_plugin.hpp"
+#include "irods_re_structs.hpp"
+#include "msParam.h"
+#include "rodsErrorTable.h"
+#include "irods_error.hpp"
+#include "irods_log.hpp"
+#include "objDesc.hpp"
+#include "fileOpr.hpp"
+
+#include <exception>
+#include <functional>
+#include <string>
+
+// persistent L1 object descriptor table
+extern l1desc_t L1desc[NUM_L1_DESC];
+
+namespace
+{
+    int msi_impl(msParam_t* _in, msParam_t* _out, ruleExecInfo_t* _rei)
+    {
+        int l1descInx = 0;
+
+        // check sanity
+        if (!_in || !_out) {
+            return SYS_INTERNAL_ERR;
```

Comment:
see previous comments

Reply:
Switched to SYS_INTERNAL_NULL_INPUT_ERR.